cms/models.py

```diff
@@ -253,6 +258,73 @@ class FacultyMembersPage(CourseProgramChildPage):
     ]
 
 
+class HomePage(Page):
+    """
+    CMS Page representing the home/root route
+    """
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    subpage_types = [
+        "CoursePage",
+        "ProgramPage",
+        "CoursesInProgramPage",
+        "LearningTechniquesPage",
+        "UserTestimonialsPage",
+        "WhoShouldEnrollPage",
+        "FacultyMembersPage",
+    ]
+
+    def _get_child_page_of_type(self, cls):
+        """Gets the first child page of the given type if it exists"""
+        child = self.get_children().type(cls).first()
+        if child:
+            return child.specific
+        return None
+
+    @property
+    def learning_experience(self):
+        """
+        Gets the "Learning Experience" section subpage
+        """
+        return self._get_child_page_of_type(LearningTechniquesPage)
+
+    @property
+    def testimonials(self):
+        """
+        Gets the testimonials section subpage
+        """
+        return self._get_child_page_of_type(UserTestimonialsPage)
+
+    @property
+    def upcoming_courseware(self):
+        """
+        Gets the upcoming courseware section subpage
+        """
+        return self._get_child_page_of_type(CoursesInProgramPage)
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        self.slug = slugify("home-{}".format(self.title))
```

Comment:
Ideally there should be only one home page, but if a user add multiple child pages under `Home Page` with same title, there will be a validation error. Do you think we should restrict users here ?

Reply:
This has to do with the slug of the home page not its children. Still, could pose a problem if someone adds another homepage with the same title. Then we're potentially looking at conflicting slugs.